### ISSUE TITLE
Can't complie plotter on centos8

### ISSUE THREAD

**Issue by ferdiina** [madMAx43v3r/chia-plotter#29]
[ 97%] Building CXX object lib/bls-signatures/src/CMakeFiles/runbench.dir/test-bench.cpp.o
**/usr/bin/ar: /usr/lib64/libgmp.so: File format not recognized**
make[2]: *** [lib/bls-signatures/src/CMakeFiles/combined_custom.dir/build.make:79: lib/bls-signatures/src/CMakeFiles/combined_custom] Error 1
[ 97%] Built target test_phase_2
make[1]: *** [CMakeFiles/Makefile2:537: lib/bls-signatures/src/CMakeFiles/combined_custom.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

Installed Packages
gmp.x86_64                                                                                    1:6.1.2-10.el8

**Comment by madMAx43v3r** [comment]
Ah I see, it's trying to use an *.so as a static library... But that's in the `bls-signatures` submodule from chia..

**Comment by ferdiina** [comment]
hi max how can i do for fix it ?

**Comment by madMAx43v3r** [comment]
Post a bug report on: https://github.com/Chia-Network/bls-signatures
They have to fix it...

**Comment by we-sell-bags** [comment]
I don't think this is their library.....
if you pull the  bls-signatures git library & follow the build instructions on the same platform then there is not this error

but if you build it from within  "chia-plotter "
then the error is thrown.

**Comment by madMAx43v3r** [comment]
Somebody managed to fix it by install a static version of libgmp and libsodium. That was on `arch` however, maybe there is something similar for centos.

**Comment by baizuchong-net** [comment]
我在centos7中并没有安装成功

**Comment by lihat1992** [comment]
Can you help me？centos8.3  Looking forward to it.

**Comment by ferdiina** [comment]
yes  i have already post issue to https://github.com/Chia-Network/bls-signatures
but just told me install python-devel or python3.x-devel
after that the issue same

**Comment by lihat1992** [comment]
I have Python3-devel installed on it, but I still have this problem.

**Comment by we-sell-bags** [comment]
it's not python.......
it's missing static libraries......

it can finally compile with this.:
try first one & if still broken use second one.

sudo yum install gmp-static 
sudo yum install  libsodium-static

Also it is critical to have the CORRECT version of "devtoolset" on your cent os

**Comment by NickHumphrey** [comment]
on Centos 8 I was able to get libsodium-static installed, and I installed gmp static with these files included:
/usr/lib64/libgmp.so
/usr/local/lib/libgmp.a

But I get:
/usr/bin/ar: /usr/lib64/libgmp.so: File format not recognized
make[2]: *** [lib/bls-signatures/src/CMakeFiles/combined_custom.dir/build.make:80: lib/bls-signatures/src/CMakeFiles/combined_custom] Error 1
make[1]: *** [CMakeFiles/Makefile2:567: lib/bls-signatures/src/CMakeFiles/combined_custom.dir/all] Error 2
make: *** [Makefile:150: all] Error 2

**Comment by madMAx43v3r** [comment]
latest master `./make_devel.sh` has usage of GMP disabled

**Comment by smash-necros** [comment]
> it's not python.......
> it's missing static libraries......
> 
> it can finally compile with this.:
> try first one & if still broken use second one.
> 
> sudo yum install gmp-static
worked for me. centos 7

**Comment by madMAx43v3r** [comment]
should be fixed now
